python/src/typed/mod.rs

```diff
@@ -126,36 +155,130 @@ pub fn default_for_member(
                 Some(_) => eyre::bail!(
                     "there should be only a single default value for non-sequence types"
                 ),
-                None => default_for_basic_type(t),
+                None => default_for_nestable_type(t, package_name, messages)?,
             },
-            NestableType::NamedType(name) => {
+            NestableType::NamedType(_) => {
                 if m.default.is_some() {
                     eyre::bail!("default values for nested types are not supported")
                 } else {
-                    let referenced_message = package_messages
-                        .get(&name.0)
-                        .context("unknown referenced message")?;
+                    default_for_nestable_type(t, package_name, messages)?
+                }
+            }
+            NestableType::NamespacedType(_) => {
+                default_for_nestable_type(t, package_name, messages)?
+            }
+        },
+        MemberType::Array(array) => match &m.default.as_deref() {
+            Some([]) => eyre::bail!("empty default value not supported"),
+            Some([default]) => preset_default_for_basic_type(&array.value_type, &default)
+                .with_context(|| format!("failed to parse default value for `{}`", m.name))?,
+            Some(_) => {
+                eyre::bail!("there should be only a single default value for non-sequence types")
+            }
```

Comment:
Why are empty arrays and arrays with multiple entries invalid as default values?

Reply:
I have fixed this issue in e55be5c79d431bb4d9fb2ca46a8f7480677a6bad